addons/membership/membership.py

```diff
@@ -136,6 +137,11 @@ def _state(self, cr, uid, ids, name, args, context=None):
             res[line.id] = state
         return res
 
+    def _set_state(self, cr, uid, ids, field_name, field_value, args, context=None):
+        cr.execute('UPDATE membership_membership_line '\
+                       'SET state=%s '\
+                       'WHERE id IN %s', (field_value, tuple([ids]),))
+        return True
```

Comment:
Well... can't we try to find another way to compute the field `state` that takes refunded invoices in account + correct dependencies on invoice object so that it is updated accordingly?